pdfgen.c

```diff
@@ -1240,6 +1278,38 @@ int pdf_add_bookmark(struct pdf_doc *pdf, struct pdf_object *page, int parent,
     return obj->index;
 }
 
+int pdf_add_link(struct pdf_doc *pdf, struct pdf_object *page, float x,
+                 float y, float width, float height,
+                 struct pdf_object *target_page, float target_y)
```

Comment:
For consistency, should we have target_x as well & add it in to the `/Dest [%u 0 R /XYZ 0 %f null]\r\n` output?

Reply:
Completely reasonable.
I had deliberately left that off to simplify the use of the functionality, and I considered that the (crystal ball) most people would be interested in just the y coordinate, but it doesn't hurt much to add in the x and makes it more generally useful.
I'll do it now.